pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -658,7 +658,7 @@ self: super: {
   }));
 
   # Need newer versions of their dependencies than the ones we have in LTS-11.x.
-  cabal2nix = super.cabal2nix.overrideScope (self: super: { hpack = addTestToolDepend (self.hpack_0_28_2) self.hspec-discover; hackage-db = self.hackage-db_2_0_1; });
+  cabal2nix = appendCompletions ["cabal2nix"] (super.cabal2nix.overrideScope (self: super: { hpack = addTestToolDepend (self.hpack_0_28_2) self.hspec-discover; hackage-db = self.hackage-db_2_0_1; }));
```

Comment:
Please note that `cabal2nix` (and some other tools) have similar overrides already in `pkgs/top-level/all-packages.nix` and elsewhere, so we have to make sure those are removed as part of this PR:

~~~
$ grep -R -- --bash-completion-script pkgs
pkgs/development/haskell-modules/hackage-packages.nix:         $exe --bash-completion-script $exe >$out/share/bash-completion/completions/stack
pkgs/top-level/all-packages.nix:      $exe --bash-completion-script $exe >$out/share/bash-completion/completions/${drv.pname}
~~~